### ISSUE TITLE
telegraf doesn't run on OpenWRT 22.03.2 arm64

### ISSUE THREAD

**Issue by nirojan** [influxdata/telegraf#12084]
### Relevant telegraf.conf

```toml
No config needed to trigger this runtime error
```

### Logs from Telegraf

```text
panic: runtime error: index out of range [1] with length 1

goroutine 1 [running]:
modernc.org/libc/honnef.co/go/netdb.init.0()
        /go/pkg/mod/modernc.org/libc@v1.16.7/honnef.co/go/netdb/netdb.go:760 +0x4c8
```

### System info

Telegraf 1.24.2/1.23.4, OpenWRT 22.03.2 (64 bit) running on Raspberry Pi 4 Model B

### Steps to reproduce

1. Download latest Linux arm64 binary (https://dl.influxdata.com/telegraf/releases/telegraf-1.24.2_linux_arm64.tar.gz) or install `telegraf-full` package from OpenWRT package manager. 
2. Run `telegraf --version`

### Expected behavior

Print telegraf version and exit

### Actual behavior

It throws a panic message. Adding `--debug` flag doesn't show any additional info.

```shell
root@router:/tmp/telegraf-1.24.2/usr/bin# ./telegraf --debug --version
panic: runtime error: index out of range [1] with length 1

goroutine 1 [running]:
modernc.org/libc/honnef.co/go/netdb.init.0()
        /go/pkg/mod/modernc.org/libc@v1.16.7/honnef.co/go/netdb/netdb.go:760 +0x4c8
```

### Additional info

The smaller build of telegraf with limited set of input, output & no Processors & aggregators that is available in OpenWrt repo doesn't have this issue. More info on what is shipped with smaller telegraf build https://openwrt.org/docs/guide-user/services/telegraf

OpenWrt repo ships telegraf 1.23.4 at the moment.

The full telegraf build provided by OpenWrt and the upstream binary for Linux arm64 suffer the same issue.

It runs fine on Ubuntu 20.04 running on Raspberry Pi 4. So the issue seems to be limited to OpenWRT.

OpenWRT issue report for this https://github.com/openwrt/packages/issues/19663

**Comment by powersj** [comment]
> /go/pkg/mod/modernc.org/libc@v1.16.7/honnef.co/go/netdb/netdb.go:760 +0x4c8

`modernc.org/libc` is an indirect dependency due to `modernc.org/sqlite`. [Here is the specific source line](https://gitlab.com/cznic/libc/-/blob/master/honnef.co/go/netdb/netdb.go#L760) that the error message references.

Looks like it is trying to parse `/etc/services` at the time. What is the difference between that file on OpenWRT and Ubuntu?
Do you also know if this worked on previous builds and suddenly broke at a recent release? 

> The smaller build of telegraf with limited set of input, output & no Processors & aggregators that is available in OpenWrt repo doesn't have this issue.

This is probably because only the ras and sql inputs & outputs use this library, which I think are omitted in that smaller build.

**Comment by nirojan** [comment]
> Looks like it is trying to parse `/etc/services` at the time. What is the difference between that file on OpenWRT and Ubuntu? 

At an overview, Ubuntu's services file has far more services listed than OpenWrt does.

```shell
❯ wc -l services-openwrt
174 services-openwrt
❯ wc -l services-ubuntu
417 services-ubuntu
```

On further digging the OpenWrt services file had a service listed without a protocol. This was what was causing the library to panic as removing the whole line or adding protocol to it stops the go library from panicking.

```
privoxy	8118
```

I presume this was added by `privoxy` package since I didn't touch this file.

Taking a look at the link you provided, there is a commit from 7 months ago to stop it from panicking when parsing the services file https://gitlab.com/cznic/libc/-/commit/bef4a02c7d76ed2c1b56053610ce107412108130

Looking at dependency file (https://github.com/influxdata/telegraf/blob/v1.24.2/go.mod#L418), the version being used is `v1.16.7` whereas the latest is `v1.21.1` https://gitlab.com/cznic/libc/-/tags.

Thank you for your help in troubleshooting this issue. I am happy for this issue to be closed.

**Comment by powersj** [comment]
> Thank you for your help in troubleshooting this issue. I am happy for this issue to be closed.

Not a problem. Could you confirm from one of our [nightly images](https://github.com/influxdata/telegraf/blob/master/docs/NIGHTLIES.md), that using the latest actually resolves this?

**Comment by nirojan** [comment]
Unfortunately nightly build still fails to run if the services file contain a service without protocol.

```shell
root@router:/tmp/telegraf-1.25.0/usr/bin# ./telegraf --version
panic: runtime error: index out of range [1] with length 1

goroutine 1 [running]:
modernc.org/libc/honnef.co/go/netdb.init.0()
        /go/pkg/mod/modernc.org/libc@v1.20.3/honnef.co/go/netdb/netdb.go:760 +0x4c8

# With the problematic line in the file
root@router:/tmp/telegraf-1.25.0/usr/bin# cat /etc/services | grep -i privoxy
privoxy 8118
```

The nightly build version I used for testing `Telegraf 1.25.0-76d7a954 (git: master@76d7a954)`

This is the services file from OpenWrt. Had to add txt file extension, so GitHub would let me upload it [services.txt](https://github.com/influxdata/telegraf/files/9841485/services.txt)

**Comment by powersj** [comment]
I've filed https://gitlab.com/cznic/libc/-/issues/25 to track this upstream. As soon as they fix and tag a new release I'll grab the updated library.

What repo can I file an issue for privoxy?

**Comment by powersj** [comment]
Alright upstream is fixed, and cznic/libc v1.21.2 has the fix to not panic. I merged #12090 which updates our version of the dependency.

Tomorrow's nightly images or our next minor release, next week, will include the fix.

I also put up https://github.com/openwrt/packages/pull/19694 to update the Makefile for that package, but I have not dug into how to test this.
